src/tasks/components/NewTaskScheduler/TaskScheduler.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {FC, ChangeEvent, useCallback, useEffect, useState} from 'react'
+
+// Components
+import {
+  Button,
+  ButtonShape,
+  ButtonType,
+  Columns,
+  ComponentColor,
+  Form,
+  Grid,
+  IconFont,
+  Input,
+  Page,
+  SelectGroup,
+} from '@influxdata/clockface'
+import TaskSchedulerFormField from 'src/tasks/components/NewTaskScheduler/TaskSchedulerFormField'
+import ScriptSelector from 'src/tasks/components/NewTaskScheduler/ScriptSelector'
+
+// Utils
+import {debouncer} from 'src/dataExplorer/shared/utils'
+import {SafeBlankLink} from 'src/utils/SafeBlankLink'
+
+// Types
+import {RemoteDataState, TaskOptions, TaskSchedule} from 'src/types'
+
+import 'src/tasks/components/NewTaskScheduler/TaskScheduler.scss'
+
+const getScripts = require('src/client/scriptsRoutes').getScripts
+
+interface Props {
+  taskOptions: TaskOptions
+  onChangeScheduleType: (schedule: TaskSchedule) => void
+  onChangeInput: (e: ChangeEvent<HTMLInputElement>) => void
+}
+
+const TaskScheduler: FC<Props> = ({
+  taskOptions,
+  onChangeScheduleType,
+  onChangeInput,
+}) => {
+  const [loading, setLoading] = useState(RemoteDataState.NotStarted)
+  const [scripts, setScripts] = useState([])
+  const [searchTerm, setSearchTerm] = useState('')
+  const [selectedScript, setSelectedScript] = useState<any>({})
+  const [taskName, setTaskName] = useState('')
+
+  const handleGetScripts = useCallback(async (name: string = '') => {
+    try {
+      setLoading(RemoteDataState.Loading)
+      const resp = await getScripts({query: {limit: 250, name}})
```

Comment:
we're moving away from making calls to the routes layer directly in components. This logic should get moved to an api layer, similar to how the identity stuff works: https://github.com/influxdata/ui/blob/master/src/identity/apis/account.ts, https://github.com/influxdata/ui/blob/master/src/identity/apis, and also a thunk layer, since it's dispatching actions to the redux state, like here: https://github.com/influxdata/ui/blob/49952435971518aedcd14c3b1b5a3c92713190cd/src/identity/actions/thunks/index.ts#L61

Making a request across the network and then translating the response into something the application can use should be the responsibility of the api layer. The application (like redux, and components) shouldn't be aware of response status codes. That's for the api layer to handle. Notice how the api files in `identity` translate error codes (`401`, `500`, `404` into named error types `UnauthorizedError`, `ServerError`, `NotFound`). This decouples the application code from the code to handle network responses.

If you need help with this, you can pair with me or Bill to understand the reasoning.